doc/src/manual/parallel-computing.md

```diff
@@ -1,12 +1,546 @@
+threads. Described as a fork-join approach, all threads have to join together and serial execution continues.
+Multi-threading is supported using the `Base.Threads` module that is still considered experimental, as Julia is
+not fully thread-safe, consider for example [this issue](https://github.com/JuliaLang/julia/issues/22581).
+Also multi-threading should be implemented only if you take into consideration global variables, locks and 
+atomics, so we will present it later.
+
+In the end we will present Julia's way to distributed and parallel computing. With scientific computing 
+in mind, Julia natively implements interfaces to send a process through multiple cores or machines.
+Also we will mention useful external packages for distributed programming like `MPI.jl` and `DistributeArrays.jl`.
+
+
+# Coroutines
+
+Julia's parallel programming platform uses [Tasks (aka Coroutines)](@ref man-tasks) to switch among multiple
+computations. Whenever code performs a communication operation like [`fetch`](@ref) or [`wait`](@ref),
+the current task is suspended and a scheduler picks another task to run. A task is restarted when
+the event it is waiting for completes.
+
+For many problems, it is not necessary to think about tasks directly. However, they can be used
+to wait for multiple events at the same time, which provides for *dynamic scheduling*. In dynamic
+scheduling, a program decides what to compute or where to compute it based on when other jobs
+finish. This is needed for unpredictable or unbalanced workloads, where we want to assign more
+work to processes only when they finish their current tasks.
+
+As an example, consider computing the singular values of matrices of different sizes:
+
+```julia-repl
+julia> M = Matrix{Float64}[rand(800,800), rand(600,600), rand(800,800), rand(600,600)];
+
+julia> pmap(svd, M);
+```
+
+If one process handles both 800×800 matrices and another handles both 600×600 matrices, we will
+not get as much scalability as we could. The solution is to make a local task to "feed" work to
+each process when it completes its current task. For example, consider a simple [`pmap`](@ref)
+implementation:
+
+```julia
+function pmap(f, lst)
+    np = nprocs()  # determine the number of processes available
+    n = length(lst)
+    results = Vector{Any}(n)
+    i = 1
+    # function to produce the next work item from the queue.
+    # in this case it's just an index.
+    nextidx() = (global i; idx=i; i+=1; idx)
+    @sync begin
+        for p=1:np
+            if p != myid() || np == 1
+                @async begin
+                    while true
+                        idx = nextidx()
+                        idx > n && break
+                        results[idx] = remotecall_fetch(f, p, lst[idx])
+                    end
+                end
+            end
+        end
+    end
+    results
+end
+```
+
+[`@async`](@ref) is similar to [`@spawn`](@ref), but only runs tasks on the local process. We
```

Comment:
Maybe a longer form explanation of `@spawn` would be good? Since the manual so-far has not explained `@spawn`.